### ISSUE TITLE
Device stopped sending updates

### ISSUE THREAD

**Issue by Harry79** [fhempy/fhempy#162]
Hi,

I have one single tuya device in my home, which was working fine with fhempy. On the 1st of June the device stopped sending updates.

I  noticed there is a new version of your library, so I upgraded. Still got no updates.

Now I am on version 0.1.656

I deleted my device and searched for devices again with the "tuya_system", one device was found; I called create_device and got the following output in fhem:
```
Failed to execute function Set: Traceback (most recent call last):
  File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/fhem_pythonbinding.py", line 385, in handle_function
    ret = await self.execute_function(hash, fhem_reply_done, nmInstance)
  File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/fhem_pythonbinding.py", line 486, in execute_function
    fct_timeout,
  File "/usr/lib/python3.7/asyncio/tasks.py", line 416, in wait_for
    return fut.result()
  File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/generic.py", line 208, in Set
    return await utils.handle_set(self._conf_set, self, hash, args, argsh)
  File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/utils.py", line 338, in handle_set
    return await fct_call(hash, final_params)
  File "/opt/fhem/.local/lib/python3.7/site-packages/fhempy/lib/tuya/tuya.py", line 307, in set_create_device
    f"tuya_local_{dev['device_id']} localkey {dev['key']}",
KeyError: 'key'
```

Still the device was created but remained in the "attr localkey required" state.

So I added the attribute localkey. Some readings were then populated but only the static stuff, like ip, lat, lon, local_key, ... But not the readings which show the current state and change over time.

Btw. independent of what I set in the attribute localkey, the reading local_key is updated to the value I used to have when the system was working.

Thanks in advance!
Harald

**Comment by fhempy** [comment]
Hi,
can you please try to remove the device from SmartLife and reset the device itself. After that add the device as a new device to SmartLife and do another scan in tuya_system.

I also fixed the localkey handling on create_device which caused the error in the stacktrace.

Thanks for reporting! Please let me know if it works after the reset procedure.

**Comment by Harry79** [comment]
Hi,

So I reset the device, added it again on https://eu.iot.tuya.com, deleted my device in fhem, searched for devices, got a new localkey, and called create_device. Now I get all the readings again. 

Thanks!
Harald

PS: Just a side question: Why was it necessary to reset my device?

**Comment by fhempy** [comment]
Actually I don't know. I just read on other forums that that happens sometimes. Good to know that it works again!

**Comment by Harry79** [comment]
It stopped working again. Yesterday at 18:53 (GMT+2) was the last update.

**Comment by fhempy** [comment]
What's type of device is it? Model?

**Comment by Harry79** [comment]
PG-300

**Comment by fhempy** [comment]
Can you post a product link? I couldn't find anything with PG-300 tuya.

**Comment by Harry79** [comment]
https://www.lidl.de/p/parkside-balkonkraftwerk-pbkw-300-a1-smart-starterset/p100363619

**Comment by fhempy** [comment]
Unfortunately I have no other idea than resetting the device again and see if it works longer. You might also try tuya_cloud, but tuya_cloud also stops sometimes as tuya cloud services are not that reliable.

Does the device work in SmartLife when it stops working in fhempy?

**Comment by Harry79** [comment]
tuya.com shows these logs:

```
  2023-06-17 18:59:23	Report	Inverter temperature	51.10℃	device itself	
  2023-06-17 18:59:23	Report	Total power	0.06kW	device itself	
  2023-06-17 18:59:23	Report	Output	CGcAAZQAADcB9Q==	device itself	
  2023-06-17 18:58:22	Report	Total power	0.06kW	device itself	
  2023-06-17 18:58:22	Report	Output	CDYAAZQAADgB9Q==	device itself	
  2023-06-17 18:56:53	Report	Inverter temperature	52.10℃	device itself	
  2023-06-17 18:56:51	Online	Online		device itself	
  2023-06-17 18:56:40	Offline	Offline	closed	device itself	
  2023-06-17 18:56:39	Report	Inverter temperature	52.10℃	device itself
  2023-06-17 18:53:37	Report	Total power	0.06kW	device itself	
  2023-06-17 18:53:37	Report	Output	CFQAAawAAEAB9Q==	device itself	
  2023-06-17 18:52:36	Report	Total power	0.06kW	device itself	
```

in fhem this was the last updated Reading
```
dp_07 | CFQAAawAAEAB9Q== | 2023-06-17 18:53:37
```

Seems like the device went offline and after it went only again fhem did not recover.

**Comment by fhempy** [comment]
Can you please try to reproduce it by disconnecting the inverter from power for a few minutes and plugging it in afterwards again?

**Comment by Harry79** [comment]
I deleted the device in fhem and re-created it, now it works since 1 1/2 hours.

Can we add some more verbose logging to see what actually is causing the issue?

**Comment by Harry79** [comment]
FHEM works for a month and then stops updating. After all this may not be an fhempy issue. As far as I got it fhempy is built upon tinytuya. Had an again no updates in fhem and looked into tinytuya. Tinytuya could also not get data from my device! It outputted "{u'Payload': None, u'Err': u'905', u'Error': u'Network Error: Device Unreachable'}". The device still responded to ping and the tuya cloud still got updates. I tried updating fhem, the rpi, restarting the rpi, pushing the reset button on the device, nothing helped. In the end I disconnected the power cable for 15 minutes after sunset. After plugging it in again it worked for tinytuya. After restarting the fhem module also fhem got the updates. So in the end it is probably a device issue.
